microceph/api/disks.go

```diff
@@ -44,3 +56,45 @@ func cmdDisksPost(s *state.State, r *http.Request) response.Response {
 
 	return response.EmptySyncResponse
 }
+
+// cmdDisksDelete is the handler for DELETE /1.0/disks/{osdid}.
+func cmdDisksDelete(s *state.State, r *http.Request) response.Response {
+	var osd string
+	osd, err := url.PathUnescape(mux.Vars(r)["osdid"])
+	if err != nil {
+		return response.InternalError(err)
+	}
+
+	var req types.DisksDelete
+	osdid, err := strconv.ParseInt(osd, 10, 64)
+
+	if err != nil {
+		return response.BadRequest(err)
+	}
+	err = json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.InternalError(err)
```

Comment:
Same as above - I think this is a client error and not an internal server side error when the body cannot be JSON decoded.